custom-elements/custom-elements-registry/construct.html

```diff
@@ -0,0 +1,77 @@
+<!DOCTYPE html>
+<title>Custom Elements: Constructor Tests</title>
+<link rel="help" href="https://html.spec.whatwg.org/multipage/dom.html#elements-in-the-dom">
+<meta name="author" title="Dominic Cooney" href="mailto:dominicc@chromium.org">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/custom-elements-helpers.js"></script>
+<body>
+<script>
+'use strict';
+
+test_with_window((w) => {
+  assert_throws(TypeError.prototype, () => {
+    w.HTMLElement();
+  }, 'calling the HTMLElement constructor should throw a TypeError');
+}, 'HTMLElement constructor, invoke');
+
+test_with_window((w) => {
+  assert_throws(TypeError.prototype, () => {
+    new w.HTMLElement();
+  }, 'invoking the HTMLElement constructor with a construct call should ' +
+     'throw a TypeError');
+}, 'HTMLElement constructor, construct');
+
+test_with_window((w) => {
+  class X extends w.HTMLElement {}
+  w.customElements.define('a-a', X);
+  assert_throws(TypeError.prototype, () => {
+    X();
+  }, 'calling a custom element constructor should throw a TypeError');
+}, 'Custom element constructor, invoke');
+
+test_with_window((w) => {
+  var num_constructor_invocations = 0;
+  class C extends w.HTMLElement {
+    constructor() {
+      super();
+      num_constructor_invocations++;
+    }
+  }
+  w.customElements.define('a-a', C);
+  let e = new C();
```

Comment:
I was wondering should this new operation throw `TypeError` in step 3 of [HTMLConstructor](https://html.spec.whatwg.org/multipage/dom.html#htmlconstructor). because `C` wasn't defined in __current global object__'s registry, but in iframe's registry?